### PR TITLE
pools: add default label for master/worker pool

### PR DESCRIPTION
**Pull request by yuqi-zhang**
Today the master/worker MCP does not have any different labels.
This means that if e.g. a user wants to add a kubeletconfig, they
must first create a label for the pool and target the kubeletconfig
against the label. This PR creates a default label for master/worker
pools to skip that step.

### CODE REVIEW

**Comment by cgwalters** [comment]
Hmm, the kubeletconfig can't target a name directly?  It seems weird to add labels that just match our names.

**Comment by yuqi-zhang** [comment]
> Hmm, the kubeletconfig can't target a name directly? It seems weird to add labels that just match our names.
by design it searches the MCPs for a label that it wants to match

2 notes:
1. it seems that our labels are all quoted (`"machineconfiguration.openshift.io/mco-built-in": ""` instead of `machineconfiguration.openshift.io/mco-built-in: ""`), whereas many other repos are not. Was there a reason for this?

2. this only considers master/worker mcp and does not consider custom pools, which isn't synced like master/worker and probably needs a minor rework if we want to do that too

**Comment by yuqi-zhang** [comment]
To be more specific on my previous point, I mean that we use labels to select the pool here: https://github.com/openshift/machine-config-operator/blob/master/pkg/controller/kubelet-config/kubelet_config_controller.go#L607 . It's not a direct kubeletconfig-pool matching. Would you say we should redesign that?

**Comment by yuqi-zhang** [comment]
/test e2e-aws-proxy

Just doing this for an environment, please ignore

**Comment by runcom** [comment]
> 3 notes:
> 
> 1. it seems that our labels are all quoted (`"machineconfiguration.openshift.io/mco-built-in": ""` instead of `machineconfiguration.openshift.io/mco-built-in: ""`), whereas many other repos are not. Was there a reason for this?

uhm, I can't remember anything there :thinking: is it causing any specific issue?

> 2\. this only considers master/worker mcp and does not consider custom pools, which isn't synced like master/worker and probably needs a minor rework if we want to do that too

I think we can target custom pools as well as we're growing further support as first-class citizen (stub ignition creation and the like)

**Comment by yuqi-zhang** [comment]
Adding it as a template instead, so it should get applied to all pools. Tested with master/worker, will test with custom pools to make sure

File: install/0000_80_machine-config-operator_01_machineconfigpool.crd.yaml [hunk 1]
```diff
@@ -69,6 +69,10 @@ spec:
     # shortNames allow shorter string to match your resource on the CLI
     shortNames:
     - mcp
+  template:
+    metadata:
+      labels:
+          pools.operator.machineconfiguration.openshift.io/{{ status.configuration.name }}: ""
```
Comment (runcom) [inline install/0000_80_machine-config-operator_01_machineconfigpool.crd.yaml:75]: omg niceee

**Comment by yuqi-zhang** [comment]
I can confirm that this works for installs/upgrades for master/worker pools, however the template doesn't seem to apply for custom pools by default. Maybe I'm mis-understanding the templating mechanism?

File: install/0000_80_machine-config-operator_01_machineconfigpool.crd.yaml (hunk 1 above)
Comment (deads2k) [reply install/0000_80_machine-config-operator_01_machineconfigpool.crd.yaml:75]: > omg niceee

hmm..... What do you think this does?

Comment (yuqi-zhang) [reply install/0000_80_machine-config-operator_01_machineconfigpool.crd.yaml:75]: Having a default label applied to each MCP object being created, do you know what the correct way to do this is?

**Comment by runcom** [comment]
alrighty, the master/worker MCP are still something managed by MCO and it makes sense to have default labels applied. Custom pools on the other hand are user-provided and it makes sense to follow the pattern in the default pool manifests (maybe we'll find a way to do that). This is an optimization for the default pools anyway, the old behavior is untouched as per acceptance criteria in the jira card

**Comment by runcom** [comment]
@openshift/openshift-team-mco-maintainers ptal

/approve

**Comment by yuqi-zhang** [comment]
I've changed it back to just adding a label for master/workers based on their manifests for now. In the long term we may want to rework the controller to manage the label

File: manifests/master.machineconfigpool.yaml [hunk 2]
```diff
@@ -5,6 +5,7 @@ metadata:
   labels:
     "operator.machineconfiguration.openshift.io/required-for-upgrade": ""
     "machineconfiguration.openshift.io/mco-built-in": ""
+    pools.operator.machineconfiguration.openshift.io/master: ""
```
Comment (ericavonb) [inline manifests/master.machineconfigpool.yaml:8]: for consistency, and because I don't trust yaml, should we put the key in quotes like the other labels?

**Comment by yuqi-zhang** [comment]
Added quotes to labels. Although I feel like we should remove them like other openshift repos, for now lets keep them consistent

**Comment by ericavonb** [comment]
> To be more specific on my previous point, I mean that we use labels to select the pool here: https://github.com/openshift/machine-config-operator/blob/master/pkg/controller/kubelet-config/kubelet_config_controller.go#L607 . It's not a direct kubeletconfig-pool matching. Would you say we should redesign that?

Wait, KubeletConfigs specify MachineConfigPools via label selectors to create MachineConfigs selected by that pool via label selectors? 😱  Isn't this backwards?

Except that it sorta assumes this'll work based on the pool name. A lot of indirection going on. KubeletConfigs should not be aware of MachineConfigPools at all, nor should they rely on any sort of naming or labeling, and they definitely shouldn't be aware of templates or rendering logic. Gah, this deserves an issue at least

**Comment by yuqi-zhang** [comment]
We can add that as part of the KCC/CRCC rework items. What would you recommendation be in terms of hooking up KC/CRC to MCs?

**Comment by yuqi-zhang** [comment]
This PR just aims to add a default. We can always remove it if we rework the behaviour

**Comment by ericavonb** [comment]
> This PR just aims to add a default. We can always remove it if we rework the behaviour

👍 sorry, yes this PR seems fine by me. 

Shouldn't have polluted the PR like that! Sorry again.

/lgtm

**Comment by yuqi-zhang** [comment]
No worries at all! Thanks for the review :D

**Pull request merged**